test/unit/registration.js

```diff
@@ -49,3 +65,94 @@ exports['bool expr is ignored (returns true) if not a string or empty'] = functi
     test.equals(false, transition.isBoolExprFalse({}, ' \t\n '));
     test.done();
 };
+
+exports['add_patient trigger creates a new patient'] = function(test) {
+    var change = { doc: {
+        form: 'R',
+        patient_id: '05649',
+        reported_date: 53,
+        from: '+555123',
+        fields: { patient_name: 'jack' }
+    } };
+    var view = sinon.stub().callsArgWith(3, null, { rows: [ { doc: { parent: { _id: 'papa' } } } ] });
+    var db = { medic: { view: view } };
+    var saveDoc = sinon.stub().callsArgWith(1);
+    var audit = { saveDoc: saveDoc };
+    var eventConfig = {
+        form: 'R',
+        events: [ { name: 'on_create', trigger: 'add_patient' } ]
+    };
+    sinon.stub(config, 'get').returns([ eventConfig ]);
+    sinon.stub(transition, 'validate').callsArgWith(2);
+    var getRegistrations = sinon.stub(utils, 'getRegistrations').callsArgWith(1, null, []);
+    transition.onMatch(change, db, audit, function() {
+        test.equals(getRegistrations.callCount, 1);
+        test.equals(getRegistrations.args[0][0].id, '05649');
+        test.equals(view.callCount, 1);
+        test.equals(view.args[0][0], 'medic-client');
+        test.equals(view.args[0][1], 'people_by_phone');
+        test.deepEqual(view.args[0][2].key, [ '+555123' ]);
+        test.equals(view.args[0][2].include_docs, true);
+        test.equals(view.args[0][0], 'medic-client');
+        test.equals(saveDoc.callCount, 1);
+        test.equals(saveDoc.args[0][0].name, 'jack');
```

Comment:
I don't think I am. The one below checks that a custom field name can be given as a parameter. This one is checking the default name ("patient_name")